app/controllers/posts_controller.rb

```diff
@@ -6,5 +6,21 @@ def index
 
   def show
     @posts = [Post.find(params[:id])]
+    @comment = Comment.new
+    @like = Like.new
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text))
```

Comment:
- [OPTIONAL ] Regarding the parameters here I recommend moving them to a method and making it private to make them strong parameters you can check this [Lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.) to know more about it 🔥🙏🏻✨